compiler/rustc_const_eval/src/interpret/intern.rs

```diff
@@ -154,6 +168,8 @@ pub fn intern_const_alloc_recursive<'tcx, M: CompileTimeMachine<'tcx, const_eval
     intern_kind: InternKind,
     ret: &MPlaceTy<'tcx>,
 ) -> Result<(), InternResult> {
+    let mut disambiguator = DisambiguatorState::new();
```

Comment:
This is just a number, right? So it's cheap to create even when we don't need it (like for non-`static` consts)?

Reply:
it's a hash map, but that's also cheap if it never gets pushed to

Reply:
Makes sense. A perf run would probably still be a good idea given how this changes a bunch of stuff all over the compiler and some of that could easily be hot code.